lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -1355,6 +1344,48 @@ def compute_timestamp(resource_type, record, time)
         ts_secs = timestamp.tv_sec
         ts_nanos = timestamp.tv_nsec
       end
+      ts_secs = begin
+                  Integer ts_secs
+                rescue ArgumentError, TypeError
+                  ts_secs
+                end
+      ts_nanos = begin
+                   Integer ts_nanos
+                 rescue ArgumentError, TypeError
+                   ts_nanos
+                 end
+
+      # Adjust timestamps from the future.
+      if ts_secs.is_a?(Integer) && ts_nanos.is_a?(Integer)
+        parsed_timestamp = Time.at(ts_secs + ts_nanos / 1_000_000_000.0)
```

Comment:
This is already computed as `timestamp` in many of the cases above. Would it make sense to assign `timestamp` in the remaining cases based on `ts_secs` and `ts_nanos` and just use `timestamp` here?

Reply:
You can explicitly set `timestamp` to `nil` in those cases, can't you, and simply skip the adjustment?

Reply:
Yeah, that works. Changed.